modules/core/src/shaderlib/picking/picking.js

```diff
@@ -3,20 +3,63 @@ import {picking, createModuleInjection} from '@luma.gl/core';
 createModuleInjection('picking', {
   hook: 'vs:DECKGL_FILTER_COLOR',
   injection: `
+  if (picking_uDrawZ) {
+    picking_setZ(geometry.position.z, geometry.pickingColor);
+  } else {
     picking_setPickingColor(geometry.pickingColor);
+  }
 `
 });
 
 createModuleInjection('picking', {
   hook: 'fs:DECKGL_FILTER_COLOR',
   order: 99,
   injection: `
+  if (picking_uDrawZ) {
+    color = picking_getZColor();
+  } else {
     // use highlight color if this fragment belongs to the selected object.
     color = picking_filterHighlightColor(color);
 
     // use picking color if rendering to picking FBO.
     color = picking_filterPickingColor(color);
-  `
+  }
+`
 });
 
-export default picking;
+export default {
+  ...picking,
+  dependencies: ['project'],
+  getUniforms: opts => {
+    const uniforms = picking.getUniforms(opts);
+    if (opts && opts.pickingActive !== undefined) {
+      uniforms.picking_uDrawZ = Boolean(opts.pickingActive && opts.pickingDrawZ);
+    }
+    return uniforms;
+  },
+  vs: `\
+const vec3 bitPackShift = vec3(1.0 / 65025.0, 1.0 / 255.0, 1.0);
```

Comment:
`65025.0`? Not `65535.0`? Add a comment?

Also, I think there is a `pack` shader module that does this.

Reply:
`255*255`

Reply:
256*256-1?